src/gemm_api.cpp

```diff
@@ -104,28 +111,33 @@ miopen_tensile_matrix transpose(const miopen_tensile_matrix& a)
-                                                         get_data_type(b), 
-                                                         get_data_type(c), 
-                                                         get_data_type(c), 
-                                                         is_transposed(a) ? a.lens[0] : a.lens[1], 
-                                                         is_transposed(b) ? b.lens[1] : b.lens[0], 
-                                                         is_transposed(a) ? a.lens[1] : a.lens[0],
-                                                         batch, 
-                                                         get_ld(a),
-                                                         a.batch.stride, 
-                                                         get_ld(b),
-                                                         b.batch.stride, 
-                                                         get_ld(c),
-                                                         c.batch.stride,
-                                                         get_ld(c),
-                                                         c.batch.stride,
-                                                         1.0);
+        auto problem = Tensile::ContractionProblem::GEMM_Strides(is_transposed(a), 
+                                                                 is_transposed(b), 
+                                                                 get_data_type(a), 
+                                                                 get_data_type(b), 
+                                                                 get_data_type(c), 
+                                                                 get_data_type(c), 
+                                                                 is_transposed(a) ? a.lens[0] : a.lens[1], 
+                                                                 is_transposed(b) ? b.lens[1] : b.lens[0], 
+                                                                 is_transposed(a) ? a.lens[1] : a.lens[0],
+                                                                 batch, 
+                                                                 get_ld(a),
+                                                                 a.batch.stride, 
+                                                                 get_ld(b),
+                                                                 b.batch.stride, 
+                                                                 get_ld(c),
+                                                                 c.batch.stride,
+                                                                 get_ld(c),
+                                                                 c.batch.stride,
+                                                                 1.0);
+
+        if (a.type == miopen_tensile_type_bfloat16 || a.type == miopen_tensile_type_int8x4)
```

Comment:
I believe you may also want to do this for `half`.

Reply:
Thank you Scott! Will do.

Reply:
done